tests/integrationtest/t/privilege/privileges.test

```diff
@@ -491,4 +494,314 @@ GRANT CREATE TEMPORARY TABLES,DELETE,EXECUTE,INSERT,SELECT,SHOW VIEW,TRIGGER,UPD
+SELECT * FROM information_schema.cluster_hardware;
+-- error 1227
+SELECT * FROM information_schema.cluster_info;
+-- error 1227
+SELECT * FROM information_schema.cluster_load;
+-- error 1227
+SELECT * FROM information_schema.cluster_systeminfo;
+-- error 1227
+SELECT * FROM information_schema.cluster_log WHERE time BETWEEN '2021-07-13 00:00:00' AND '2021-07-13 02:00:00' AND message like '%';
+disconnect ccnobody;
+
+connect (ccconfig,localhost,ccconfig,,);
+connection ccconfig;
+SHOW GRANTS;
+SELECT * FROM information_schema.cluster_config limit 0;
+SELECT * FROM information_schema.cluster_HARDWARE limit 0;
+-- error 1227
+SELECT * FROM information_schema.cluster_INFO;
+-- error 1227
+SELECT * FROM information_schema.cluster_LOAD;
+-- error 1227
+SELECT * FROM information_schema.cluster_SYSTEMINFO;
+-- error 1227
+SELECT * FROM information_schema.cluster_LOG WHERE time BETWEEN '2021-07-13 00:00:00' AND '2021-07-13 02:00:00' AND message like '%';
+disconnect ccconfig;
+
+connect (ccprocess,localhost,ccprocess,,);
+connection ccprocess;
+SHOW GRANTS;
+SELECT * FROM information_schema.CLUSTER_info limit 0;
+SELECT * FROM information_schema.CLUSTER_load limit 0;
+SELECT * FROM information_schema.CLUSTER_systeminfo limit 0;
+SELECT * FROM information_schema.CLUSTER_log WHERE time BETWEEN '1970-07-13 00:00:00' AND '1970-07-13 02:00:00' AND message like '%';
+-- error 1227
+SELECT * FROM information_schema.CLUSTER_config;
+-- error 1227
+SELECT * FROM information_schema.CLUSTER_hardware;
+disconnect ccprocess;
+connection default;
+
+
+# TestSetPasswdStmt
+CREATE USER 'superuser';
+CREATE USER 'nobodyuser';
+GRANT ALL ON *.* TO 'superuser';
+connect (superuser,localhost,superuser,,);
+connection superuser;
+SET PASSWORD for 'nobodyuser' = 'newpassword';
+SET PASSWORD for 'nobodyuser' = '';
+disconnect superuser;
+
+connect (nobodyuser,localhost,nobodyuser,,);
+connection nobodyuser;
+--error Fail
+SET PASSWORD for 'superuser' = 'newpassword';
+disconnect nobodyuser;
+connection default;
+
+
+# TestShowGrantsWithRolesAndDynamicPrivs
+CREATE ROLE tsg_r1;
+CREATE USER tsg_u1, tsg_u2;
+GRANT CONNECTION_ADMIN, ROLE_ADMIN, SYSTEM_VARIABLES_ADMIN, PROCESS ON *.* TO tsg_r1;
+GRANT CONNECTION_ADMIN ON *.* TO tsg_u1 WITH GRANT OPTION;
+GRANT CONNECTION_ADMIN ON *.* TO tsg_u2 WITH GRANT OPTION;
+GRANT ROLE_ADMIN ON *.* TO tsg_u1;
+GRANT ROLE_ADMIN ON *.* TO tsg_u2;
+GRANT ROLE_ADMIN ON *.* TO tsg_r1 WITH GRANT OPTION;
+GRANT CONFIG ON *.* TO tsg_r1;
+GRANT tsg_r1 TO tsg_u1, tsg_u2;
+SET DEFAULT ROLE tsg_r1 TO tsg_u1;
+
+connect (tsg_u1,localhost,tsg_u1,,);
+connection tsg_u1;
+SHOW GRANTS;
+SHOW GRANTS FOR CURRENT_USER();
+SHOW GRANTS FOR 'tsg_u1';
+disconnect tsg_u1;
+
+connect (tsg_u2,localhost,tsg_u2,,);
+connection tsg_u2;
+SHOW GRANTS;
+SHOW GRANTS FOR CURRENT_USER();
+SHOW GRANTS FOR 'tsg_u2';
+SET ROLE tsg_r1;
+SHOW GRANTS;
+SHOW GRANTS FOR CURRENT_USER();
+SHOW GRANTS FOR 'tsg_u2';
+disconnect tsg_u2;
+connection default;
+
+
+# TestGetEncodedPassword
```

Comment:
Seems we cannot port this case directly.

Reply:
Why not?

https://github.com/pingcap/tidb/pull/47292/files#diff-8b507d8e9d7f506b565cbedfd8e7c860aa76099537d36ed9cdeaf931d71d64c0L1171

https://github.com/pingcap/tidb/pull/47292/files#diff-d305408c493422010fcbfbf7a67ed9e3ccdd0161d3eb5c74797535e13ab50a53R548-R550